src/Payments.sol

```diff
@@ -296,24 +424,47 @@ contract Payments is
         Account storage payer = accounts[rail.token][rail.from];
         Account storage payee = accounts[rail.token][rail.to];
 
+        uint256 oldRate = rail.paymentRate;
+
+        // For terminated rails, can only reduce rate, not increase it
+        if (rail.terminationEpoch > 0) {
+            require(
+                newRate <= oldRate,
+                "failed because terminated at epoch: cannot increase rate on terminated rail"
+            );
+            require(
+                oneTimePayment <= rail.lockupFixed,
+                "failed because terminated at epoch: one-time payment exceeds fixed lockup"
+            );
+        }
+
         OperatorApproval storage approval = operatorApprovals[rail.token][
             rail.from
         ][rail.operator];
-        require(approval.isApproved, "Operator not approved");
 
-        // Settle the payer's lockup.
-        // If we're changing the rate, we require full settlement.
+        // Settle the payer's lockup to account for elapsed time
         (bool fullySettled, uint256 lockupSettledUpto) = settleAccountLockup(
             payer
         );
-        if (newRate != rail.paymentRate) {
-            require(
-                fullySettled && lockupSettledUpto == block.number,
-                "account lockup not fully settled; cannot change rate"
-            );
-        }
 
-        uint256 oldRate = rail.paymentRate;
+        // If we're increasing the rate, we require full account lockup setlement before we can proceed
+        require(
+            newRate <= oldRate ||
+                (fullySettled && lockupSettledUpto == block.number),
```

Comment:
Can it be `fullySettled` and not `lockupSettledUpto == block.number`?

Reply:
no this happens. I've simplified the `settleAccountLockup` function to return only one value now.